transcom/mymove
# B 19976 int

## [19976](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A949508)

## Summary

This task adds the ability for the prime to add a standaloneCrate option to the Domestic Crating item. Its functionality is nearly identical to a standard crating item, with the caveat that when the TOO goes to approve the item, the highest dollar amount that can be approved is 1000$. The TOO approval is part of a follow-on task and is not completed here.

Code removal is from the previous solution for 19976 that added an additional service item. Code had been merged into integration, so it had to be removed here.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

Prime UI:
1. Access the Office application
2. Log in to the Prime Simulator
3. Select 'Add a Service Item'
4. Select to add a Domestic Crate from the Service Item Type dropdown
5. Select domestic crating from the next dropdown.
6. Check the Standalone Crate option
7. Complete the form
8. Submit the form
9. Verify that the Service Item was added with a standalone tag in the DB.

Swagger:
PayLoad: {"item":{"length":4,"width":4,"height":4},"crate":{"length":5,"width":5,"height":5},"moveTaskOrderID":"1035d1f9-fe75-498f-98b3-21788d3084c3","mtoShipmentID":"5b600a58-944b-49ed-ba99-390886ab3d3e","modelType":"MTOServiceItemDomesticCrating","standaloneCrate":true,"reason":"i want to","description":"moped","reServiceCode":"DCRT"}

URL:http://officelocal:3000/swagger-ui/prime.html

Endpoint: mto-service-items

1. Update the supplied payload with an existing moveTaskOrderID and mtoShipmentID from your environment that is available to the prime.
2. Submit the payload to the provided endpoint.
3. Verify in the DB that the service item was added with a standalone tag.

### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
![Screenshot 2024-05-21 at 7 47 01 PM](https://github.com/transcom/mymove/assets/162157577/f5d9010f-1d29-46b0-a3d4-ad867d2a4cf6)

![Screenshot 2024-05-21 at 7 44 38 PM](https://github.com/transcom/mymove/assets/162157577/f2d138d1-d44b-468a-8f1a-2b195f19c00f)
